crates/table/src/layout.rs

```diff
@@ -424,13 +425,25 @@ impl From<SumType> for SumTypeLayout {
             .collect::<Vec<_>>()
             .into();
 
-        let tag_offset = max_child_size as u16;
-        let size = align_to(max_child_size + 1, max_child_align) as u16;
-        let align = max_child_align as u16;
+        // Guarantees that tag fits inside align.
+        let align = u16::max(max_child_align as u16, 1);
+
+        // Ensure the payload field is sufficiently aligned for all its members.
+        // `max_child_size` and `max_child_align` will already be consistent
+        // if the most-aligned variant is also the largest,
+        // but this is not necessarily the case.
+        // E.g. if variant A is a product of 31 `u8`s, and variant B is a single `u64`,
+        // `max_child_size` will be 31 and `max_child_align` will be 8.
+        // Note that `payload_size` may be 0.
+        let payload_size = align_to(max_child_size, max_child_align);
```

Comment:
Wait, on further reflection, maybe this isn't necessary? Because in that case, it's fine for the payload to be size 31, right?